Improvement: Search Feature for help expensify

studentofcoding opened the pull request:
As per the last discussion with @marcochavezf [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1691519742649079?thread_ts=1691499129.775699&cid=C01GTK53T8Q) we are using XML Configuration to handle the Search Feature

- Make related components & add gse script
- Configure CSP for each inline script
- Styling related components
- Add relevant setup (Annotations, & Context)

Additional context: 
- The initial state of https://help.expensify.com/ is having errors on CSP. This PR handles that also. 
- There's an unrelated error from Google inline script that is related to ads `async-ads.js`. It's not affected the functionality & already hides the ads anyway.

<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/23978
PROPOSAL: https://github.com/Expensify/App/issues/23978#issuecomment-1659762731


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Open https://help.expensify.com/
2. Click Search Icon to open the Search sidebar
3. On the Opened Search Sidebar Type the query you want to find & Press "Enter" or Click Magnifying glass icon
4. If there are results, They'll be shown on the list below it.
5. If there aren't, It'll show `Upps no results found, Please try again...`.
6. To Clear the result click the X button on the search input.
7. To Close the Search sidebar, click the X button on the right side top of the Sidebar

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Open https://help.expensify.com/
2. Click Search Icon to open the Search sidebar
3. On the Opened Search Sidebar Type the query you want to find & Press "Enter" or Click Magnifying glass icon
4. If there are results, They'll be shown on the list below it.
5. If there aren't, It'll show `Upps no results found, Please try again...`.
6. To Clear the result click the X button on the search input.
7. To Close the Search sidebar, click the X button on the right side top of the Sidebar

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>
<img width="1440" alt="Screenshot 2023-08-11 at 05 07 42" src="https://github.com/Expensify/App/assets/20473526/fb04e841-8cce-4221-ad6c-7748119a81f2">

https://www.loom.com/share/892b56bba57f463a8c839a2b9a712ee5?sid=d2dfe9d4-35f8-4c67-935d-aaaccb6a4f51



</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/20473526/7fee3456-adff-4ec1-9636-8ac56e0c7f59



</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/20473526/c9f78c4e-cac5-4bbd-8b0f-d94bcde19ea5



</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

studentofcoding committed:
Setup Search Engine for help Expensify with XML Configuration

- Make related components
- Configure CSP for each inline script
- Styling related components
- Add relevant setup (Annotations, & Context)

studentofcoding committed:
Make Sidebar full width up until table breakpoint

studentofcoding committed:
Add CSP for Captcha

studentofcoding commented:
I have read the CLA Document and I hereby sign the CLA

studentofcoding commented:
recheck

studentofcoding commented:
_I have read the CLA Document and I hereby sign the CLA_

studentofcoding committed:
Fix prettier error

marcochavezf commented:
@eVoloshchak, let me know if you have bandwidth to review it soon. I can jump into the review tomorrow

marcochavezf commented:
@studentofcoding thank you for raising the PR! Just one more thing, can you upload videos (instead of screenshots) of the search functionality in the `screenshots/videos` section? It would be required for design review since it involves different UI changes

studentofcoding commented:
> @studentofcoding thank you for raising the PR! Just one more thing, can you upload videos (instead of screenshots) of the search functionality in the `screenshots/videos` section? It would be required for design review since it involves different UI changes

The videos are added @marcochavezf, thanks!

marcochavezf commented on docs/_includes/sidebar-search.html:
```diff
@@ -0,0 +1,13 @@
+        <div class="gcse-searchbox"></div>
+        <div class="gcse-searchresults"></div>
+    </div>
+</div>
```
```suggestion
</div>

```

marcochavezf commented on docs/_includes/search-toggle.html:
```diff
@@ -0,0 +1,5 @@
+<div class="search-icon">
+    <a href="javascript:void(0)" onclick="openSidebar()" id="toggle-search-open">
```
I think the href is not necessary here
```suggestion
    <a onclick="openSidebar()" id="toggle-search-open">
```

marcochavezf commented on docs/_includes/sidebar-search.html:
```diff
@@ -0,0 +1,13 @@
+    <div id="sidebar-search" class="sidebar">
+        <div class="flex searchbar-title-wrapper">
+            <h4 class="search-title">Search</h4>
+            <a href="javascript:void(0)" id="toggle-search-close">
```
```suggestion
            <a id="toggle-search-close">
```

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,45 @@ function injectFooterCopywrite() {
     footer.innerHTML = `&copy;2008-${new Date().getFullYear()} Expensify, Inc.`;
 }
 
+// Count property of --scroll-y variable
+window.addEventListener('scroll', () => {
+    document.documentElement.style.setProperty('--scroll-y', `${window.scrollY}px`);
+});
```
Let's move this listened inside of `DOMContentLoaded` event https://github.com/Expensify/App/blob/01fc9285e5eeff2ac79173d799fad75f9d624d99/docs/assets/js/main.js#L78

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,45 @@ function injectFooterCopywrite() {
+function openSidebar() {
+    document.getElementById('sidebar-layer').style.display = 'block';
+    document.getElementById('gsc-i-id1').placeholder = 'Search for something...';
+    // Make body unscrollable
```
Add a space between a comment an a block of code
```suggestion

    // Make body unscrollable
```

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,45 @@ function injectFooterCopywrite() {
+
+function closeSidebar() {
+    document.getElementById('sidebar-layer').style.display = 'none';
+    // Make body scrollable again
```
```suggestion

    // Make body scrollable again
```

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,45 @@ function injectFooterCopywrite() {
     footer.innerHTML = `&copy;2008-${new Date().getFullYear()} Expensify, Inc.`;
 }
 
+// Count property of --scroll-y variable
```
Update the comment to "why" we're doing this

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,45 @@ function injectFooterCopywrite() {
+    const scrollY = body.style.top;
+    body.style.position = '';
+    body.style.top = '';
+    window.scrollTo(0, parseInt(scrollY || '0', 10) * -1);
```
Add a little comment about why we're scrolling to the "Y" value in `scrollTo` and what represents that value

marcochavezf commented on docs/context.xml:
```diff
@@ -0,0 +1,33 @@
+    <sort_by_keys label="Relevance" key=""></sort_by_keys>
+    <sort_by_keys label="Date" key="date"></sort_by_keys>
+    <autocomplete_settings source="1"></autocomplete_settings>
+</CustomSearchEngine>
```
```suggestion
</CustomSearchEngine>

```

marcochavezf reviewed (changes requested):
Looks great! I just left a few comments to change a few minor things, but everything looks good so far 🚀

studentofcoding committed:
Update docs/context.xml

Co-authored-by: Marco Chávez <marcochavezf@gmail.com>

studentofcoding committed:
Update all

studentofcoding committed:
Update

studentofcoding commented:
All resolved @marcochavezf

shawnborton commented:
A couple of comments:

It looks like all of the content is cut off at the bottom.

Can you double check that the correct font is being applied to the smaller text under the links?

For the search button, we should just use our standard green button styles here. Same for the text input, can we make sure this matches what we have on NewDot?

studentofcoding committed:
Use NewDot search button, font, & fix cut off content on search result

studentofcoding commented:
> It looks like all of the content is cut off at the bottom.
> 
> Can you double check that the correct font is being applied to the smaller text under the links?
> 
> For the search button, we should just use our standard green button styles here. Same for the text input, can we make sure this matches what we have on NewDot?

Done it all @shawnborton, here are the vercel staging for references https://xml-docs.vercel.app

cc : @marcochavezf

shawnborton commented:
Are the screenshots updated?

studentofcoding commented:
> Are the screenshots updated?

Yup all is updated

studentofcoding committed:
Make the Label Animation work and adjustment

shawnborton commented:
For this UI:
<img width="225" alt="image" src="https://github.com/Expensify/App/assets/2319350/ba26d032-1421-4cc9-bb81-b46bc77157dd">

What does the green magnifying glass do? Do you only get search results when you press enter or press the icon?

We shouldn't be using the icon like this though, we should use our standard green button. We can put an icon inside of the button though. 

@marcochavezf are there existing mockups that we're working off of?

studentofcoding commented:
Yes, as this is inline script from google, how it works is via Enter Key or Clicking the Magnifying Glass icon thus I think we should keep the icon there @shawnborton 

On the Button change, could you share with me any ref of it?

studentofcoding committed:
Update Search button

studentofcoding commented:
> This is what our buttons look like: <img alt="image" width="399" src="https://user-images.githubusercontent.com/2319350/259749627-288ce316-76ff-47c7-868c-152fd5e8a8e3.png">

What about now @shawnborton?
<img width="1440" alt="Screenshot 2023-08-10 at 20 17 21" src="https://github.com/Expensify/App/assets/20473526/f5636be7-e8f8-474d-823f-4506b841a1f5">

studentofcoding committed:
Make the icon same size on mobile

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,190 @@ button {
+}
+
+.search-title {
+    font-size: 120%;
```
Why are we using 120% for the title? We use 17px in NewDot for these headers, let's standardize on that.

studentofcoding replied:
Noted, done it

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,190 @@ button {
+    left: 0;
+    right: 0;
+    bottom: 0;
+    background-color: #1b574470;
```
What is this color?

studentofcoding replied:
This is for the background layer here @shawnborton 
<img width="1440" alt="Screenshot 2023-08-10 at 21 37 17" src="https://github.com/Expensify/App/assets/20473526/eb5d6b61-da14-4908-916d-b2a699eca5c2">

studentofcoding replied:
This is changed to `background-color: rgba(0, 0, 0, 0.4);` as per NewDot design

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,190 @@ button {
+    font-family: "ExpensifyNeue", "Segoe UI Emoji", "Noto Color Emoji" !important;
+}
+
+#gsc-iw-id1 {
```
What is this element?

studentofcoding replied:
Yupp comment added

shawnborton replied:
The comment isn't very specific though. For example, what is this particular ID for? A search box? Something else?

shawnborton commented on docs/assets/images/close.svg:
This isn't an icon coming from our own system. We already have a close icon in the NewDot repo.

studentofcoding replied:
I can't find it, could you help reference it @shawnborton, thanks!

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,190 @@ button {
+
+#sidebar-search {
+    background-color: $color-green-appBG;
+    width: 400px;
```
Where does 400 come from? Again, I think we use 375px on NewDot for the right hand sidepane. 

Also, what does this look like on mobile?

studentofcoding replied:
Yess, It's fixed now

shawnborton commented:
The search input + button should look something like this:
<img width="805" alt="image" src="https://github.com/Expensify/App/assets/2319350/ef148803-d5b5-4d1f-a06c-aad8bd290e61">

The button is at 40x40, aligned to the baseline of the input. The icon inside of the button is at 16x16

The top header area does not use a "x" button, but rather just uses a back arrow to close. This is consistent with the rest of the NewDot app.

studentofcoding commented on docs/_sass/_main.scss:
```diff
@@ -524,6 +709,13 @@ button {
+}
+
+.base-icon--small {
+    width: 14px;
```
Change to only use base-icon

studentofcoding committed:
Change icon to back left, Alignment styling, & remove console.log

studentofcoding commented:
Done it all @shawnborton

studentofcoding commented:
Also the video are updated @shawnborton

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+    background-color: $color-green-appBG;
+    border-bottom: rgba(255, 255, 255, 0.2) 2px solid;
+    border-bottom-left-radius: 0px;
+    &:focus-within {
```
Missing blank line above

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+}
+
+.gsc-search-button.gsc-search-button-v2 svg {
+    fill: white;
```
Rather than white, this should be our text color.

studentofcoding replied:
Done

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+    background-color: $color-green-appBG;
+    border: $color-green-appBG;
+    font-family: "ExpensifyNeue", "Helvetica Neue", "Helvetica", Arial, sans-serif !important;
+    max-height: 80vh;
```
What exactly does this do?

studentofcoding replied:
This is to make the search result list didn't crop out as you mention here
https://github.com/Expensify/App/pull/24299#issuecomment-1671913061

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+    display: block;
+    position: absolute;
+    margin-top: -20px;
+    font-size: 15px;
```
Does this exist as a variable already? Maybe not...

studentofcoding commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+
+input#gsc-i-id1.gsc-input {
+    background-color: $color-green-appBG;
+    color: #E7ECE9;
```
Yes, it is var from `colors.scss` $color-green-appBG: #061B09;

Changed to use `$color-appBG` as the default

shawnborton replied:
Looks like the color is still not a var.

studentofcoding commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+    left: 0;
+    right: 0;
+    bottom: 0;
+    background-color: rgba(0, 0, 0, 0.4);
```
Yess it is, below is the reference
<img width="1440" alt="Screenshot 2023-08-10 at 22 53 00" src="https://github.com/Expensify/App/assets/20473526/5570ac78-26cf-4f48-9c49-48d570a9cfa6">

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+    transform: translateY(-100%) scale(0.80);
+}
+
+label {
```
Should we give this more specificity? Otherwise it will impact all labels, not just those specific to google search.

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+}
+
+/* Manage Google Search label animation */
+input:focus+label,
```
Same comment as below, should we give this more specificity?

studentofcoding replied:
Done refactored it into 
```
input#gsc-i-id1:focus+label.search-label,
input#gsc-i-id1:valid+label.search-label,
input#gsc-i-id1:active+label.search-label
```

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,195 @@ button {
+}
+
+/* Hide the relevance, etc sections */
+.gsc-above-wrapper-area {
```
You can group all of these classes together, like:

```
.gsc-above-wrapper-area,
.gsc-webResult.gsc-result .gsc-url-top,
etc {
    display: none;
}
```

studentofcoding replied:
Ahh yes, changed it

studentofcoding commented:
Noted, are there any other else @shawnborton ?

studentofcoding committed:
Refactor & Tidyup

studentofcoding committed:
Fix Prettier diff

studentofcoding commented:
Is there any other feedback on this @marcochavezf

marcochavezf commented:
@studentofcoding thanks for addressing the comments, let's wait for a final UI review. Meanwhile can you update the videos with the latest changes?

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,180 @@ button {
+    /* Specify a stack order in case you're using a different order for other elements */
+}
+
+/* All gsc id & class are Google Search related */
```
This is still not specific at all though. What exactly is this element? is it an input? is it a button? something else?

studentofcoding replied:
Noted, the recent one is already updated @shawnborton. Please let me know what you think of it!

shawnborton commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,180 @@ button {
+
+#gsc-iw-id1 {
+    background-color: $color-appBG;
+    border-bottom: rgba(255, 255, 255, 0.2) 2px solid;
```
We have a border color in our colors file we can use for this.

studentofcoding replied:
Noted, done

shawnborton commented:
Just left a couple more comments. @marcochavezf can you review as well please?

marcochavezf commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,180 @@ button {
+    font-family: "ExpensifyNeue", "Segoe UI Emoji", "Noto Color Emoji";
+    transform: translateY(-50%);
+    left: 20px;
+    color: rgb(175, 187, 176);
```
Let's change this color to hexadecimal and move it to a variable in _main.css (or reuse the color if it's duplicated)

studentofcoding replied:
Noted

marcochavezf commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,180 @@ button {
+}
+
+.gsc-search-button.gsc-search-button-v2:hover {
+    background-color: rgb(0, 168, 98);
```
Also this one

studentofcoding replied:
Noted

studentofcoding commented on docs/_sass/_main.scss:
```diff
@@ -182,6 +183,180 @@ button {
 }
 
+.logo {
+    width: 80%;
```
Ahh yes, this is leftover, thanks for reminding @shawnborton

marcochavezf commented on docs/_sass/_main.scss:
Given the complexity of the styles for the search bar, can you move them to a new file and import it in _main.scss?

studentofcoding replied:
Yes, would `_search-bar.scss` suit the needs? @marcochavezf

studentofcoding replied:
`_search-bar.scss` is now created !

studentofcoding committed:
Make search-bar styling individual & Manage svg based on screen size

studentofcoding commented:
All is updated @marcochavezf @shawnborton, and Videos are updated!

shawnborton commented:
I checked the video and it looks like the input slightly jumps 1-2px when the cursor focuses:
<img width="265" alt="image" src="https://github.com/Expensify/App/assets/2319350/bdd1ff3b-f030-4ce9-8a7c-1440216540f5">


Any ideas?

studentofcoding commented:
> I checked the video and it looks like the input slightly jumps 1-2px when the cursor focuses:
> 
> <img width="265" alt="image" src="https://github.com/Expensify/App/assets/2319350/bdd1ff3b-f030-4ce9-8a7c-1440216540f5">
> 
> 
> 
> 
> 
> Any ideas? 

Ahh yes, It's should be static now, thanks @shawnborton

studentofcoding committed:
Fix moving input

studentofcoding commented:
> I checked the video and it looks like the input slightly jumps 1-2px when the cursor focuses: <img alt="image" width="265" src="https://user-images.githubusercontent.com/2319350/259850131-bdd1ff3b-f030-4ce9-8a7c-1440216540f5.png">
> 
> Any ideas?

The Jump is fixed & videos are updated @shawnborton 

cc: @marcochavezf

shawnborton commented:
Great, thank you! Last thing, and sorry for not catching this sooner, but the close icon should be using our standard icon color (I think it's in the color vars file?) and on hover, it can just use our standard text color.
<img width="51" alt="image" src="https://github.com/Expensify/App/assets/2319350/d59df0b2-837c-41a2-a030-48f5d7f01b69">

studentofcoding commented:
> Great, thank you! Last thing, and sorry for not catching this sooner, but the close icon should be using our standard icon color (I think it's in the color vars file?) and on hover, it can just use our standard text color. <img alt="image" width="51" src="https://user-images.githubusercontent.com/2319350/259880976-d59df0b2-837c-41a2-a030-48f5d7f01b69.png">

No Problemo @shawnborton, The fix is otw!

studentofcoding committed:
Fix on hover close icon

studentofcoding commented:
> Great, thank you! Last thing, and sorry for not catching this sooner, but the close icon should be using our standard icon color (I think it's in the color vars file?) and on hover, it can just use our standard text color. <img alt="image" width="51" src="https://user-images.githubusercontent.com/2319350/259880976-d59df0b2-837c-41a2-a030-48f5d7f01b69.png">

The Hover is fixed @shawnborton

shawnborton commented:
Thank you! @marcochavezf feel free to do the final review.

studentofcoding commented:
> Thank you! @marcochavezf feel free to do the final review.

Thank you too @shawnborton !

studentofcoding commented:
Let's go @marcochavezf

I'm so excited to finish it, lol!

studentofcoding commented:
> @studentofcoding thanks for addressing the comments, let's wait for a final UI review. Meanwhile can you update the videos with the latest changes?

All UI Review is done @marcochavezf 

I'll standby today to finish all the final feedback 😁

marcochavezf commented on docs/_sass/_search-bar.scss:
```diff
@@ -0,0 +1,197 @@
+@import 'colors';
+@import 'fonts';
+
+$color-appBG: $color-green-appBG;
+$color-highlightBG: $color-green-highlightBG;
+$color-accent : $color-green400;
+$color-borders: $color-green-borders;
+$color-icons: $color-green-icons;
+$color-text: $color-white;
+$color-link: $color-blue300;
+$color-link-hovered: $color-blue200;
+$color-success: $color-green400;
+$color-text-supporting: $color-light-gray-green;
+$color-green-hover: $color-green-hover;
+$color-gray-label: $color-gray-label;
```
Ah why are we reassigning the color variables to new variables? The code would be more maintainable if we user the original variables directly

studentofcoding replied:
This is the pattern we have on the codebase @marcochavezf. If we refactor it, would it be another scope then?

studentofcoding replied:
From what I see, why it's re-assigned is because some of the variables are color names 
e.g.
```
$color-text: $color-white;
$color-link: $color-blue300;
$color-link-hovered: $color-blue200;
$color-success: $color-green400;
```

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,95 @@ function injectFooterCopywrite() {
+
+    // Check if the viewport is smaller than tablet
+    if (!mediaQuery.matches) {
+        // Iterate through each SVG element
```
```suggestion
```

The code is self-explanatory here, so this comment is not necessary

studentofcoding replied:
Noted

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,95 @@ function injectFooterCopywrite() {
+    if (!mediaQuery.matches) {
+        // Iterate through each SVG element
+        Array.from(svgsGoogle).forEach((svg) => {
+            // Set the viewBox attribute to '0 0 13 13' to make the svg fit in the mobile view
```
```suggestion

            // Set the viewBox attribute to '0 0 13 13' to make the svg fit in the mobile view
```

studentofcoding replied:
Done

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,95 @@ function injectFooterCopywrite() {
+            svg.setAttribute('width', '13');
+        });
+    } else {
+        // Iterate through each SVG element
```
```suggestion
```

studentofcoding replied:
Done

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,95 @@ function injectFooterCopywrite() {
+    } else {
+        // Iterate through each SVG element
+        Array.from(svgsGoogle).forEach((svg) => {
+            // Set the viewBox attribute to '0 0 20 20' to make the svg fit in the tablet-desktop view
```
```suggestion

            // Set the viewBox attribute to '0 0 20 20' to make the svg fit in the tablet-desktop view
```

studentofcoding replied:
Done

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,95 @@ function injectFooterCopywrite() {
+    }
+}
+
+// Function to insert element after another
```
Explain instead why we're insert an element after another

studentofcoding replied:
Done

studentofcoding replied:
Done

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,95 @@ function injectFooterCopywrite() {
+    referenceNode.parentNode.insertBefore(newNode, referenceNode.nextSibling);
+}
+
+// Need to wait up until page is load, so the svg viewBox can be changed
+// And the search label can be inserted
```
👍🏽

studentofcoding replied:
Done

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,95 @@ function injectFooterCopywrite() {
+    window.scrollTo(0, parseInt(scrollY || '0', 10) * -1);
+}
+
+// Function to change SVG viewBox from Google
```
Explain why we're changing the SVG in this function

studentofcoding replied:
Done

marcochavezf commented on docs/assets/js/main.js:
```diff
@@ -75,9 +75,95 @@ function injectFooterCopywrite() {
+    body.style.top = `-${yAxis}`;
+}
+
+// Function to close the sidebar
```
```suggestion
```

The function is self-explanatory here

studentofcoding replied:
Done

studentofcoding replied:
Done

marcochavezf reviewed (changes requested):
The code is looking great! 👍🏽 Just a few other minor changes

studentofcoding committed:
Fix some comments

studentofcoding commented:
> The code is looking great! 👍🏽 Just a few other minor changes

Done it all @marcochavezf

marcochavezf commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

marcochavezf reviewed (dismissed):
LGTM, thank you @studentofcoding!

studentofcoding committed:
Fix prettier error

studentofcoding commented:
> LGTM, thank you @studentofcoding!

Can you approve it again? There is lint error before and it's fix now @marcochavezf Thanks!

studentofcoding commented:
Hmm the Jest and perf checks already taking 15 mins ++ while I only running lint fix `npm run prettier`

Do I need to do anything @marcochavezf ?

marcochavezf commented:
> Hmm the Jest and perf checks already taking 15 mins ++ while I only running lint fix `npm run prettier`
> 
> Do I need to do anything @marcochavezf ?

We'd just need to wait for the checks to pass

studentofcoding commented:
> Details

Noted

studentofcoding commented:
> > Hmm the Jest and perf checks already taking 15 mins ++ while I only running lint fix `npm run prettier`
> > Do I need to do anything @marcochavezf ?
> 
> We'd just need to wait for the checks to pass

Noted, and after that, Do I need to merge it or it'll auto Merged @marcochavezf? As I'm already at 3 AM and wonder if I still have to standby

Thanks!

studentofcoding commented:
Finally the check is done. 

If you still have time, Please kindly check approve it @marcochavezf 🙏🏻

Thanks

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5859813282) to staging by https://github.com/marcochavezf in version: 1.3.54-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

johncschuster commented:
Hey folks! I've noticed a few things that might have missed QA (or didn't explicitly get considered as "wrong")

### Body text color is dark, but it should be bright to match the body text of the rest of the help site

When you misspell a word and get a suggested replacement, the color of the text that says, "Did you mean:" is dark (almost black), but shouldn't be. It should match the text color of the light body text of the rest of the help site.

Keep in mind, we should make sure that the text color will change when we give users the option to switch between dark mode and light mode.

<img width="1500" alt="2023-08-15_12-53-12" src="https://github.com/Expensify/App/assets/5579997/d81e5f18-ca1f-4440-a7d0-26e3611881e1">


### Background color of the pagination is bright. It does not match the background color of the rest of the helpsite.

Keep in mind, we should make sure that the background color of the pagination will change when we give users the option to switch between dark mode and light mode.

<img width="1500" alt="2023-08-15_12-50-10" src="https://github.com/Expensify/App/assets/5579997/d5162cbf-2a68-477c-87c5-48a518936469">


CC @shawnborton

marcochavezf commented:
Hi @studentofcoding, while reviewing the new implementation, we noticed a few minor visual aspects (posted above by @johncschuster) that might need some attention. Can you create a follow-up PR to address those?

studentofcoding commented:
Hi, @marcochavezf sure thing!, Where do I need to connect the branch from, as I checked it's not on `main` yet. Do I need to connect it based on this PR branch?

And just a follow-up on this requirement `Keep in mind, we should make sure that the text color will change when we give users the option to switch between dark mode and light mode.`. As this is not based on the issue request, Will it extend the scope, or it's just a note and we can use the same variable color for now?

Thanks

@johncschuster

studentofcoding commented:
Noted @shawnborton, like this?

<img width="378" alt="Screenshot 2023-08-16 at 02 16 11" src="https://github.com/Expensify/App/assets/20473526/73b0c202-951f-4ec8-aa71-7ad0006e1767">

shawnborton commented:
Yup, much better for the text.

studentofcoding commented:
PR opened : https://github.com/Expensify/App/pull/24598

@marcochavezf @johncschuster @shawnborton

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5885771241) to production by https://github.com/yuwenmemon in version: 1.3.54-13 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
